source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -403,6 +403,11 @@ Http::FilterHeadersStatus JsonTranscoderFilter::encodeHeaders(Http::HeaderMap& h
     // In gRPC wire protocol, headers frame with end_stream is a trailers-only response.
     // The return value from encodeTrailers is ignored since it is always continue.
     encodeTrailers(headers);
+
+    // When there is no body, a empty JSON array is returned by default. Set the content type
+    // correctly.
+    headers.insertContentType().value().setReference(Http::Headers::get().ContentTypeValues.Json);
```

Comment:
I think encodeTrailers() call may have set the content-type in line L596

Reply:
That should be fine because it just sets it to JSON, never GRPC.